packages/rule-babel-config-is-valid/src/rule.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * @fileoverview `babel-config-is-valid` warns again providing an invalid babel configuration file.
+ */
+import * as ajv from 'ajv';
+import * as _ from 'lodash';
+
+import { Category } from 'sonarwhal/dist/src/lib/enums/category';
+import { debug as d } from 'sonarwhal/dist/src/lib/utils/debug';
+import { IRule, IRuleBuilder, IBabelConfigInvalid, IBabelConfigInvalidSchema } from 'sonarwhal/dist/src/lib/types';
+import { RuleContext } from 'sonarwhal/dist/src/lib/rule-context';
+import { RuleScope } from 'sonarwhal/dist/src/lib/enums/rulescope';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+const rule: IRuleBuilder = {
+    create(context: RuleContext): IRule {
+        /**
+         * Returns a readable error for 'additionalProperty' errors.
+         */
+        const generateAdditionalPropertiesError = (error: ajv.ErrorObject): string => {
+            const property = error.dataPath.substr(1);
+            const additionalProperty = (error.params as ajv.AdditionalPropertiesParams).additionalProperty;
+
+            return `'${property}' ${error.message}. Additional property found '${additionalProperty}'.`;
+        };
+
+        /**
+         * Returns a readable error for 'enum' errors.
+         */
+        const generateEnumError = (error: ajv.ErrorObject): string => {
+            const property = error.dataPath.substr(1);
+            const allowedValues = (error.params as ajv.EnumParams).allowedValues;
+
+            return `'${property}' ${error.message} '${allowedValues.join(', ')}'. Value found '${error.data}'`;
+        };
+
+        /**
+         * Returns a readable error for 'pattern' errors.
+         */
+        const generatePatternError = (error: ajv.ErrorObject) => {
+            const property = error.dataPath.substr(1);
+
+            return `'${property}' ${error.message.replace(/"/g, '\'')}. Value found '${error.data}'`;
+        };
+
+        const generateTypeError = (error: ajv.ErrorObject) => {
+            const property = error.dataPath.substr(1);
+
+            return `'${property}' ${error.message.replace(/"/g, '\'')}.`;
+        };
+
+        /**
+         * Returns a readable error message.
+         */
+        const prettyfy = (error: ajv.ErrorObject): string => {
+            let result: string;
+
+            switch (error.keyword) {
+                case 'additionalProperties':
```

Comment:
Try to avoid the switch. You can do something like:

```ts
const errorGenerators = [generateAdditionalPropertiesError, generateEnumError, generatePatternError, generateTypeError];

const result = errorGenerators.some(error) || error.message;
```

And then inside each function you check the content of `error.keyword` and return `null` if it isn't what is expected.